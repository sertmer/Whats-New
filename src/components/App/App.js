import React, { Component } from 'react';
import local from '../../data/local';
import entertainment from '../../data/entertainment'
import health from '../../data/health'
import science from '../../data/science'
import technology from '../../data/technology'
import './App.css';
import NewsContainer from '../NewsContainer/NewsContainer'
import Menu from '../Menu/Menu'

class App extends Component {
  constructor() {
    super();
    this.state = {
       currentPage: local
    }
  }

  changeCurrentPage = (type) => {
    this.setState({currentPage: type})
  }

  render() {
    return (
      <div className="app">
        <Menu 
          changeCurrentPage={this.changeCurrentPage}
          local={local}
          entertainment={entertainment}
          health={health}
          science={science}
          technology={technology}/>
        <NewsContainer data={this.state.currentPage}/>
      </div>
    );
  }
}

export default App;
