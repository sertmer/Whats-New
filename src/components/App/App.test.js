import React from 'react';
import App from './App';
import { shallow } from 'enzyme'


describe('App', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  it('should match the snapshot with all data passed in correctly', () => {   
    expect(wrapper).toMatchSnapshot()
  })

  it('Should update state when changeCurrentPage is invoked', () => {
    const mockEntertainment = [{
      id: 1,
      headline: 'a',
      img: 'b',
      url: 'c', 
      description: 'd'
    }]
    const mockLocal = [{
      id: 1,
      headline: 'e',
      img: 'f',
      url: 'g', 
      description: 'h'
    }]
    const expected = {currentPage: mockEntertainment}
    const defaultState = {currentPage: mockLocal}

    wrapper.setState(defaultState)
    expect(wrapper.state()).toEqual(defaultState)

    wrapper.instance().changeCurrentPage(mockEntertainment)
    expect(wrapper.state()).toEqual(expected)
  })

  it('should update state to reflect searchCurrentPage results', () => {
    const mockLocal = [{
      id: 1,
      headline: 'e',
      img: 'f',
      url: 'g', 
      description: 'h'
    },
    {
      id: 2,
      headline: 'i',
      img: 'j',
      url: 'k', 
      description: 'l'
    },
    {
      id: 3,
      headline: 'm',
      img: 'n',
      url: 'o', 
      description: 'p'
    }]
    const defaultState = {currentPage: mockLocal}
    const query = 'i'
    const expected = {currentPage: [
        {
          id: 2,
          headline: 'i',
          img: 'j',
          url: 'k', 
          description: 'l'
        }
      ]
    }

    wrapper.setState(defaultState)

    wrapper.instance().searchCurrentPage(query)
    expect(wrapper.state()).toEqual(expected)
  })

})


