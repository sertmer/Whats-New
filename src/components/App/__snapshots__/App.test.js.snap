// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<div
  className="app"
>
  <SearchForm
    searchCurrentPage={[Function]}
  />
  <main>
    <Menu
      changeCurrentPage={[Function]}
    />
    <NewsContainer
      data={Array []}
    />
  </main>
</div>
`;
